shepherd_bms/src/stateMachine.cpp

```diff
@@ -213,177 +193,84 @@ void StateMachine::requestTransition(BMSState_t next_state)
+            {.id = "High Cell Voltage",       .timer =       overVolt_tmr, .data_1 = accData->max_voltage.val, .optype_1 = GT, .lim_1 =                                       MAX_VOLT * 10000, .code =            CELL_VOLTAGE_TOO_HIGH,   .data_2 = digitalRead(CHARGE_DETECT), .optype_2 = EQ, .lim_2 =             HIGH }, 
+            {.id = "High Temp",               .timer =       highTemp_tmr, .data_1 =    accData->max_temp.val, .optype_1 = GT, .lim_1 =                                          MAX_CELL_TEMP, .code =                     PACK_TOO_HOT   /* -----------------------------------UNUSED---------------------------------*/  }, 
+            {.id = "Extremely Low Voltage",   .timer =        lowCell_tmr, .data_1 = accData->min_voltage.val, .optype_1 = LT, .lim_1 =                                                    900, .code =                 LOW_CELL_VOLTAGE   /* -----------------------------------UNUSED---------------------------------*/  }, 
 
+            NULL
+        };
 
-	//prefault for DCL
 
-	if (prefaultOverCurr.faultEvalState == BEFORE_TIMER_START && (accData->pack_current) > ((accData->discharge_limit + DCDC_CURRENT_DRAW)*10*1.04)) // *104% to account for current sensor +/-A
+	uint32_t fault_status = 0;
+    int incr = 0;
+
+    while (&fault_table[incr] != NULL)
     {
-		prefaultOverCurr.faultTimer.startTimer(PRE_OVER_CURR_TIME);
-		prefaultOverCurr.faultEvalState = DURING_FAULT_EVAL;
-    }
-    else if (prefaultOverCurr.faultEvalState == DURING_FAULT_EVAL)
-    {
-        if (prefaultOverCurr.faultTimer.isTimerExpired())
-        {
-			if (prefaultCANDelay2.isTimerExpired())
-			{
-            	compute.sendDclPreFault(true);
-				prefaultCANDelay2.startTimer(CAN_MESSAGE_WAIT);
-			}
-        }
-        if (!((accData->pack_current) > ((accData->discharge_limit + DCDC_CURRENT_DRAW)*10*1.04)))
-        {
-            prefaultOverCurr.faultTimer.cancelTimer();
-            prefaultOverCurr.faultEvalState = BEFORE_TIMER_START;
-        }
+        fault_status |= faultEval(fault_table[incr]);
+        incr++;
     }
 
+    return fault_status;
+    
+}
 
-	if (overCurr.faultEvalState == BEFORE_TIMER_START && (accData->pack_current) > ((accData->discharge_limit + DCDC_CURRENT_DRAW)*10*1.04)) // *104% to account for current sensor +/-A
-    {
-		overCurr.faultTimer.startTimer(OVER_CURR_TIME);
-		overCurr.faultEvalState = DURING_FAULT_EVAL;
-    }
-    else if (overCurr.faultEvalState == DURING_FAULT_EVAL)
-    {
-        if (overCurr.faultTimer.isTimerExpired())
-        {
-            faultStatus |= DISCHARGE_LIMIT_ENFORCEMENT_FAULT;
-        }
-        if (!((accData->pack_current) > ((accData->discharge_limit + DCDC_CURRENT_DRAW)*10*1.04)))
-        {
-            overCurr.faultTimer.cancelTimer();
-            overCurr.faultEvalState = BEFORE_TIMER_START;
-        }
-    }
+uint32_t StateMachine::faultEval(fault_eval index)
+{
+    bool param1;
```

Comment:
Can we make these names less generic? param feels weird. Even like "condition1" would be a bit better